docs/openapi-fetch/middleware-auth.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Middleware & Auth
+---
+
+# Middleware & Auth
+
+## Middleware
+
+Middleware allows you to modify either the request, response, or both for all fetches.
+
+Each middleware can provide `onRequest()` and `onResponse()` callbacks, which can observe and/or mutate requests and responses.
+
+```ts
+import createClient from "openapi-fetch";
+import type { paths } from "./api/v1";
+
+const myMiddleware: Middleware = {
+  async onRequest(req, options) {
+    // set "foo" header
+    req.headers.set("foo", "bar");
+    return req;
+  },
+  async onResponse(res, options) {
+    const { body, ...resOptions } = res;
+    // change status of response
+    return new Response(body, { ...resOptions, status: 200 });
+  },
+};
+
+const client = createClient<paths>({ baseUrl: "https://myapi.dev/v1/" });
+
+// register middleware
+client.use(myMiddleware);
+```
+
+::: tip
+
+The order in which middleware are registered matters. For requests, `onRequest()` will be called in the order registered. For responses, `onResponse()` will be called in **reverse** order. That way the first middleware gets the first “dibs” on requests, and the final control over responses.
+
+:::
+
+### onRequest
+
+```ts
+onRequest(req, options) {
+  // …
+}
+```
+
+`onRequest()` takes 2 params:
+
+| Name      |        Type         | Description                                                                                                                                                                          |
+| :-------- | :-----------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `req`     | `MiddlewareRequest` | A standard [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) with `schemaPath` (OpenAPI pathname) and `params` ([params](/openapi-fetch/api#fetch-options) object) |
+| `options` |   `MergedOptiosn`   | Combination of [createClient](/openapi-fetch/api#create-client) options + [fetch overrides](/openapi-fetch/api#fetch-options)                                                        |
+
+And it expects either:
+
+- **If modifying the request:** A [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request)
+- **If not modifying:** `undefined` (void)
+
+### onResponse
+
+```ts
+onResponse(res, options) {
+  // …
+}
+```
+
+`onResponse()` also takes 2 params:
+| Name | Type | Description |
+| :-------- | :-----------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `req` | `MiddlewareRequest` | A standard [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response). |
+| `options` | `MergedOptiosn` | Combination of [createClient](/openapi-fetch/api#create-client) options + [fetch overrides](/openapi-fetch/api#fetch-options) |
```

Comment:
Typo. Now I'm thinking these must be generated? In which case should they really be checked in?